crates/component-echo-http/src/lib.rs

```diff
@@ -0,0 +1,90 @@
+wit_bindgen::generate!({
+    world: "component"
+});
+
+/// Implementation of the 'component' world in wit/world.wit will hang off of this struct
+struct Component;
+
+use crate::exports::wasi::http::incoming_handler::Guest;
+use crate::wasi::http::types::{
+    Fields, IncomingBody, IncomingRequest, Method, OutgoingBody, OutgoingResponse, ResponseOutparam,
+};
+use crate::wasi::io::streams::StreamError;
+
+/// Maximum bytes to read at a time from the incoming request body
+/// this value is chosen somewhat arbitrarily, and is not a limit for bytes read,
+/// but is instead the amount of bytes to be read *at once*
+const MAX_READ_BYTES: u32 = 2048;
+
+impl Guest for Component {
+    fn handle(request: IncomingRequest, outparam: ResponseOutparam) {
+        // Pull together information from the request
+        let method: String = match request.method() {
+            Method::Get => "GET".into(),
+            Method::Post => "POST".into(),
+            Method::Patch => "PATCH".into(),
+            Method::Put => "PUT".into(),
+            Method::Delete => "DELETE".into(),
+            Method::Options => "OPTIONS".into(),
+            Method::Head => "HEAD".into(),
+            Method::Connect => "CONNECT".into(),
+            Method::Trace => "TRACE".into(),
+            Method::Other(m) => m,
+        };
+
+        // Read the body
+        let Ok(incoming_req_body) = request.consume() else {
+            panic!("failed to consume incoming request body");
+        };
+        let Ok(incoming_req_body_stream) = incoming_req_body.stream() else {
+            panic!("failed to build stream from incoming request body");
+        };
+        let mut body_buf = Vec::<u8>::with_capacity(MAX_READ_BYTES as usize);
+        loop {
+            match incoming_req_body_stream.read(MAX_READ_BYTES as u64) {
+                Ok(bytes) => body_buf.extend(bytes),
+                Err(StreamError::Closed) => {
+                    body_buf.shrink_to_fit();
+                    break;
+                }
+                Err(e) => {
+                    panic!("failed to read bytes: {e}");
+                }
+            }
+        }
```

Comment:
Can we stream the body instead?

Reply:
yeah so we *could*, but the point of this component is to echo the things about the request as JSON as it's good for demoing/using in tests -- it's not really about making an efficient server that returns back what was received, at whatever size.

Reply:
Since this is no longer an `echo` component, I think we can skip adding the extra robustness yet. It's a bit more clear (and I spell out in the documentation) that this component is not meant to be a robust echo server.